packages/component-library/src/HexOverlay/HexOverlay.js

```diff
@@ -1,110 +1,102 @@
-    window.addEventListener('resize', this.resize);
-    this.resize();
+    this._animate();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.data.length !== this.props.data.length) {
+      this._animate();
+    }
   }
 
   componentWillUnmount() {
-    window.removeEventListener('resize', this.resize);
+    this._stopAnimate();
   }
 
-  onViewportChange(viewport) {
-    this.setState({
-      viewport: {...this.state.viewport, ...viewport},
-    });
+  _animate() {
+    this._stopAnimate();
+
+    // wait 1.5 secs to start animation so that all data are loaded
+    this.startAnimationTimer = window.setTimeout(this._startAnimate, 1500);
   }
 
-  resize() {
-    this.onViewportChange({
-      width: window.innerWidth > 900 ? 898 : window.innerWidth,
-      height: 450,
-    });
+  _startAnimate() {
+    this.intervalTimer = window.setInterval(this._animateHeight, 20);
   }
 
-  render() {
-    const {viewport, data} = this.state;
-    const { mapboxStyle, mapboxToken } = this.props;
-
-    return (
-      <div className={mapWrapper}>
-        <MapGL
-          {...viewport}
-          className={'MapGL'}
-          mapStyle="mapbox://styles/mapbox/dark-v9"
-          // mapStyle={mapboxStyle}
-          // onViewportChange={this.onViewportChange.bind(this)} // or
-          onViewportChange={ viewport => this.onViewportChange(viewport)}
-          mapboxApiAccessToken={mapboxToken}
-        >
-          <DeckGLOverlay viewport={viewport} data={data || []}
-                         colorScale={colorScale}
-          />
-        </MapGL>
-      </div>
-    );
+  _stopAnimate() {
+    window.clearTimeout(this.startAnimationTimer);
+    window.clearTimeout(this.intervalTimer);
   }
-}
 
-// if we use these styles it doesn't look as good as if we just hard code the one above, TODO: fix that
-HexOverlay.propTypes = {
-  mapboxStyle: PropTypes.string,
-  mapboxToken: PropTypes.string.isRequired,
-};
+  _animateHeight() {
+    if (this.state.elevationScale === elevationScale.max) {
+      this._stopAnimate();
+    } else {
+      this.setState({elevationScale: this.state.elevationScale + 1});
+    }
+  }
+
+  render() {
+    const {viewport, data, opacity, radius, elevation, colorRange, lightSettings, coverage, upperPercentile} = this.props;
+
+    if (!data) {
+      return null;
+    }
 
-HexOverlay.defaultProps = {
-  mapboxStyle: "mapbox://styles/mapbox/dark-v9",
-};
+    // const defaultProps = {
+    //   radius: 1000,
+    //   upperPercentile: 100,
+    //   coverage: 1
+    // };
 
-export default HexOverlay
+    const layers = [
+      new HexagonLayer({
+        id: 'heatmap',
+        colorRange,
+        coverage: 1,
+        data,
+        elevationRange: [0, 3000],
+        elevationScale: elevation,
+        extruded: true,
+        getPosition: d => d,
+        lightSettings: lightSettings,
+        // onHover: this.props.onHover,
+        opacity,
+        // pickable: Boolean(this.props.onHover),
```

Comment:
Why are `onHover` and `pickable` commented out? If we aren't using them, they should be deleted.